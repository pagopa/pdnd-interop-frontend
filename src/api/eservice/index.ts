export * from './eservice.queries'
export * from './eservice.mutations'
export * from './eservice.downloads'
export * from './eservice.services'
