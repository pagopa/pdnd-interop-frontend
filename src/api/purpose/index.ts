export * from './purpose.queries'
export * from './purpose.downloads'
export * from './purpose.mutations'
export * from './purpose.services'
