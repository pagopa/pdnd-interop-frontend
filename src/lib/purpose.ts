import { DecoratedPurpose, Purpose } from '../../types'
import { Location } from 'history'
import { URL_FRAGMENTS } from './constants'
import { getBits } from './router-utils'

export function decoratePurposeWithMostRecentVersion(purpose: Purpose): DecoratedPurpose {
  const sorted = purpose.versions.sort(
    (a, b) => new Date(a.createdAt).getTime() - new Date(b.createdAt).getTime()
  )
  const mostRecentVersion = sorted[sorted.length - 1]
  const currentVersion =
    mostRecentVersion.state === 'WAITING_FOR_APPROVAL' && sorted.length >= 2
      ? sorted[sorted.length - 2]
      : mostRecentVersion

  return {
    ...purpose,
    mostRecentVersion,
    currentVersion,
    awaitingApproval: mostRecentVersion.id !== currentVersion.id,
  }
}

export function getComputedPurposeState(purpose: DecoratedPurpose): Array<string> {
  const isEserviceActive =
    purpose.eservice.descriptor.state === 'PUBLISHED' ||
    purpose.eservice.descriptor.state === 'DEPRECATED'
  const isAgreementActive = purpose.agreement.state === 'ACTIVE'
  const isPurposeActive = purpose.currentVersion.state === 'ACTIVE'

  const possibleReasons = [
    { labelKey: 'eservice', outcome: isEserviceActive },
    { labelKey: 'agreement', outcome: isAgreementActive },
    { labelKey: 'purpose', outcome: isPurposeActive },
  ]

  const reasons = possibleReasons
    .map(({ outcome, labelKey }) => (!outcome ? labelKey : undefined))
    .filter((r) => r) as Array<string>

  return reasons
}

export function getPurposeFromUrl(location: Location<unknown>) {
  const bits = getBits(location)

  // If we are in edit mode, strip that path bit
  if (Object.values(URL_FRAGMENTS.EDIT).includes(bits[bits.length - 1])) {
    bits.pop()
  }

  return bits.pop()
}
