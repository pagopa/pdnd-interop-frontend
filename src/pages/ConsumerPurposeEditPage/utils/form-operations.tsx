import {
  CheckboxGroup,
  RadioGroup,
  Select,
  TextField,
} from '@/components/shared/ReactHookFormInputs'
import type { InputOption } from '@/types/common.types'
import React from 'react'
import identity from 'lodash/identity'
import type {
  Dependency,
  DynamicFormOperations,
  Questions,
  QuestionV1,
  QuestionV2,
} from '../types/risk-analysis.types'
import { RiskAnalysisSwitch } from '../components/PurposeEditStep2RiskAnalysis/RiskAnalysisSwitch'

export const dynamicFormOperationsVersions: DynamicFormOperations = {
  '1.0': {
    getUpdatedQuestions: (values, riskAnalysis) => {
      const updatedQuestions = riskAnalysis.questions.reduce((acc, next) => {
        // We are sure we can only queue the questions that come after the current one
        // because the dependencies are always on the questions that come before.
        // E.g. question 3 can only have question 1 and/or 2 as a dependency,
        // not question 4 and subsequent
        const queuedQuestions = Object.values(acc).map((d) => d.id)

        // Check the dependencies of this questions
        const satisfiesAllDependencies = next.dependencies.every((d) => {
          // If the question is a dependency
          const isInArray = queuedQuestions.includes(d.id)
          const id = d.id as string

          // and if the field currently has the same value as stated in the dependency
          const hasSameValue = Array.isArray(values[id])
            ? (values[id] as Array<unknown>).includes(d.value)
            : values[id] === d.value

          return isInArray && hasSameValue
        })

        // Then it needs to be added as a new question to ask
        if (satisfiesAllDependencies) {
          acc[next.id] = next
        }

        return acc
      }, {} as Questions)
      return updatedQuestions
    },

    buildForm: (questions, _, lang, t) => {
      const questionKeys = Object.keys(questions)

      const formComponents = questionKeys.map((id, i) => {
        const { type, label, options, infoLabel } = questions[id] as QuestionV1

        const sx = questionKeys.length - 1 === i ? { mb: 0 } : {}
        const inputOptions = options ? options.map((o) => ({ ...o, label: o.label[lang] })) : []
        const commonProps = {
          key: id,
          name: id,
          label: label[lang],
          infoLabel: infoLabel ? infoLabel[lang] : undefined,
          sx,
        }

        switch (type) {
          case 'text':
            return <TextField {...commonProps} rules={{ required: true }} />

          case 'select-one':
            return (
              <Select
                {...commonProps}
                options={inputOptions}
                emptyLabel={t('edit.step2.emptyLabel')}
                rules={{ required: true }}
              />
            )

          case 'checkbox':
            const errorMessage =
              options && options?.length > 1
                ? t('edit.step2.multiCheckboxField.validation.mixed.required')
                : t('edit.step2.singleCheckboxField.validation.mixed.required')
            return (
              <CheckboxGroup
                {...commonProps}
                options={inputOptions}
                rules={{
                  validate: (value) =>
                    (typeof value !== 'undefined' && value.length > 0) || errorMessage,
                }}
              />
            )

          case 'radio':
            return <RadioGroup {...commonProps} options={inputOptions} rules={{ required: true }} />
        }
      })

      return { formComponents: formComponents, isSubmitBtnDisabled: false }
    },
  },
  '2.0': {
    getUpdatedQuestions: (values, riskAnalysis) => {
      const questions = riskAnalysis.questions as Array<QuestionV2>

      // Filters all the questions that not satisfies the dependency inside the "dependencies" property
      const updatedQuestions = questions.filter(({ dependencies }) => {
        function satisfiesDependency(dep: QuestionV2['dependencies'][0]) {
          if (Array.isArray(values[dep.id])) {
            return (values[dep.id] as Array<unknown>).includes(dep.value)
          }
          if (Array.isArray(dep.value)) {
            return dep.value.includes(values[dep.id])
          }
          return values[dep.id] === dep.value
        }
        // If length of dependencies array is 0 the "every" method returns true
        return dependencies.every(satisfiesDependency)
      })

      // Array<Question> -> Record<"id", Question>
      return updatedQuestions.reduce((acc, next) => ({ ...acc, [next.id]: next }), {} as Questions)
    },

    buildForm: (questions, formMethods, lang, t) => {
      const values = formMethods.getValues()

      const formComponents: Array<React.ReactNode> = []

      const questionIds = Object.keys(questions)

      type QuestionOption = Exclude<QuestionV2['options'], undefined>[0]

      function checkOptionDependency(dependency: Dependency) {
        const currentDepValue = values[dependency.id]

        const hasDependency =
          currentDepValue === dependency.value ||
          (Array.isArray(currentDepValue) &&
            (currentDepValue as string[]).includes(dependency.value as string))

        return hasDependency
      }

      function parseOption(option: QuestionOption, { hideOption, id, defaultValue }: QuestionV2) {
        // if the key "hideOption" is present in the question object and the conditions are satisfied
        // the option will be not added to the array of options
        const shouldHideOption =
          hideOption &&
          hideOption[option.value] &&
          hideOption[option.value].some(checkOptionDependency)

        if (!shouldHideOption) {
          return { value: option.value, label: option.label[lang] }
        }

        if ((values[id] as string[])?.includes(option.value)) {
          formMethods.setValue(id, defaultValue)
        }
      }

      function buildFormQuestionComponents(
        { id, type, ...question }: QuestionV2,
        inputOptions: Array<InputOption>,
        isLast: boolean
      ) {
        const questionComponents: Array<React.ReactNode> = []

        const maxLength = question?.validation?.maxLength
        const isRequired = question.required
        const isMultipleChoice = question.dataType === 'multi'

        let label = question.label[lang]
        let infoLabel = question.infoLabel && question.infoLabel[lang]

        if (maxLength) {
          const maxLengthLabel = t('edit.step2.validation.maxLength', { num: maxLength })
          if (infoLabel) {
            infoLabel += `. ${maxLengthLabel}`
          } else {
            infoLabel = maxLengthLabel
          }
        }

        const labelValidation: Array<string> = []

        if (isRequired) {
          labelValidation.push(t('edit.step2.validation.required'))
        }

        if (isMultipleChoice) {
          labelValidation.push(t('edit.step2.validation.multipleChoice'))
        }

        if (labelValidation.length > 0) {
          label += ` (${labelValidation.join(', ')})`
        }

        const sx = isLast ? { mb: 0 } : {}
        const commonProps = { key: id, name: id, label, infoLabel, sx }

        switch (type) {
          case 'text':
            questionComponents.push(
              <TextField {...commonProps} inputProps={{ maxLength }} rules={{ required: true }} />
            )
            break
          case 'select-one':
            questionComponents.push(
              <Select
                {...commonProps}
                options={inputOptions}
                emptyLabel={t('edit.step2.emptyLabel')}
                rules={{ required: true }}
              />
            )
            break
          case 'checkbox':
            questionComponents.push(
              <CheckboxGroup
                {...commonProps}
                options={inputOptions}
                rules={{
                  validate: (value) =>
                    (typeof value !== 'undefined' && value.length > 0) ||
                    t('edit.step2.multiCheckboxField.validation.mixed.required'),
                }}
              />
            )
            break
          case 'radio':
            questionComponents.push(
              <RadioGroup {...commonProps} options={inputOptions} rules={{ required: true }} />
            )
            break
          case 'switch':
            questionComponents.push(
              <RiskAnalysisSwitch
                {...commonProps}
                options={inputOptions}
                rules={{
                  validate: (value) =>
                    (typeof value === 'boolean' && value === true) ||
                    t('edit.step2.riskAnalysisSwitch.validation.boolean.isValue'),
                }}
              />
            )
            break
        }

        return questionComponents
      }

      for (let i = 0; i < questionIds.length; i++) {
        const questionId = questionIds[i]

        const question = questions[questionId] as QuestionV2

        const inputOptions = (question.options
          ?.map((option) => parseOption(option, question))
          .filter(identity) || []) as Array<InputOption>

        formComponents.push(
          ...buildFormQuestionComponents(question, inputOptions, questionIds.length - 1 === i)
        )
      }

      return { formComponents, isSubmitBtnDisabled: false }
    },
  },
}

export function getFormOperations(version: string) {
  const dynamicFormOperations = dynamicFormOperationsVersions[version]

  if (!dynamicFormOperations) {
    throw new Error('There is no business logic for this specific risk analysis data form')
  }

  return dynamicFormOperations
}
