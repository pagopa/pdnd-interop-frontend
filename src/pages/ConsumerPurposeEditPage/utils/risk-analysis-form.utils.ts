import type { Dependency, FormConfigQuestion, HideOption } from '@/api/api.generatedTypes'
import type { AnswerValue, Answers, Questions } from '../types/risk-analysis-form.types'

/**
 * Transform the answers from the backend to the frontend format
 * The answer from the backend arrives as an array of strings, but the frontend
 * needs a specific format depending on the `visualType` value.
 *
 * - `switch` needs a boolean
 * - `checkbox` needs an array of strings
 * - the rest needs a string
 *
 * @param backendAnswer The answer from the backend
 * @param question The question that the answer belongs to
 * @returns The answer in the type that the frontend needs
 */
export function getFrontendAnswerValue(
  backendAnswer: Array<string>,
  visualType?: FormConfigQuestion['visualType']
): AnswerValue {
  if (visualType === 'switch') {
    return backendAnswer[0] === 'true'
  }

  if (visualType === 'checkbox') {
    return backendAnswer
  }

  return backendAnswer[0]
}

/**
 * Transform the answers from the frontend to the backend format
 * The backend always needs an array of strings.
 *
 * @param answer The answer from the frontend
 * @returns The answer in the type that the backend needs
 * */
export function getBackendAnswerValue(answer: AnswerValue): Array<string> {
  if (typeof answer === 'boolean') {
    return [answer.toString()]
  }

  if (Array.isArray(answer)) {
    return answer
  }

  if (typeof answer === 'string') {
    return [answer]
  }

  throw new Error(`Invalid answer type: ${typeof answer} \n\n ${JSON.stringify(answer, null, 2)}`)
}

/**
 * Returns the valid answers that should be sent to the backend.
 * The answers that should be sent to the backend are only the answers
 * to the questions that are currently visible to the user.
 * The answers to the questions that are not visible to the user are not sent.
 *
 * @param currentQuestionsIds - the ids of the questions that are currently visible to the user
 * @param answers - the answers to all the questions
 * @returns the valid answers that should be sent to the backend
 * */
export function getValidAnswers(currentQuestionsIds: Array<string>, answers: Answers) {
  return Object.keys(answers).reduce<Record<string, Array<string>>>((acc, key) => {
    // If the question is actually visible, add it to the valid answers
    if (currentQuestionsIds.includes(key)) {
      return { ...acc, [key]: getBackendAnswerValue(answers[key]) }
    }
    return acc
  }, {})
}

/**
 * Returns true if the dependency is satisfied, false otherwise.
 *
 * @param dependency - the dependency
 * @param answers - the actual form values
 * @returns `true` if the dependency is satisfied, `false` otherwise
 * */
export function isDependencySatisfied(dependency: Dependency | HideOption, answers: Answers) {
  const answer = answers[dependency.id]
  if (Array.isArray(answer)) {
    return answer.includes(dependency.value)
  }

  return answer === dependency.value
}

/**
 * Returns the updated question data.
 *
 * @param answers - the actual form values
 * @param riskAnalysis - the risk analysis document
 *
 * @returns the updated questions data
 * */
export function getUpdatedQuestions(
  answers: Answers,
  riskAnalysisQuestions: FormConfigQuestion[]
): Questions {
  // Filters all the questions that not satisfies the dependency inside the "dependencies" property
  return riskAnalysisQuestions.reduce<Questions>((acc, question) => {
    const doesSatisfyDeps = question.dependencies.every((dependency) =>
      isDependencySatisfied(dependency, answers)
    )
    if (doesSatisfyDeps) {
      acc[question.id] = question
    }
    return acc
  }, {})
}

/**
 * Returns the default values for the risk analysis form.
 * The default values are the values that the form should have when it is first loaded.
 * The default values are the values that the user has already answered, if any.
 * If the user has not answered any question, the values are the default values
 * defined in the risk analysis document.
 *
 * @param riskAnalysisConfig the risk analysis document
 * @param backendAnswers the answers data, taken from the Purpose
 * @returns the default values for the risk analysis form
 */
export function getRiskAnalysisDefaultValues(
  riskAnalysisConfigQuestions: FormConfigQuestion[],
  backendAnswers?: Record<string, Array<string>>
): Answers {
  return riskAnalysisConfigQuestions.reduce<Answers>((acc, question) => {
    const answer = backendAnswers?.[question.id] ?? question.defaultValue
    acc[question.id] = getFrontendAnswerValue(answer, question.visualType)

    if (!acc[question.id] && question.dataType === 'FREETEXT') {
      acc[question.id] = ''
    }

    return acc
  }, {})
}
