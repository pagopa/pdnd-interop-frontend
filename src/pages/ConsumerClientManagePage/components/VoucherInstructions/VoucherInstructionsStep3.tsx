import React from 'react'
import { EServiceQueries } from '@/api/eservice'
import { SectionContainer } from '@/components/layout/containers'
import { StepActions } from '@/components/shared/StepActions'
import { API_GATEWAY_INTEFACE_URL } from '@/config/env'
import { RouterLink } from '@/router'
import { Link, Skeleton, Stack } from '@mui/material'
import { useTranslation } from 'react-i18next'
import { useClientKind } from '@/hooks/useClientKind'
import type { VoucherInstructionsStepProps } from '../../types/voucher-instructions.types'
import { InformationContainer } from '@pagopa/interop-fe-commons'

export const VoucherInstructionsStep3: React.FC<VoucherInstructionsStepProps> = (props) => {
  const clientKind = useClientKind()
  if (clientKind === 'CONSUMER') {
    return <ClientVoucherInstructionsStep3 {...props} />
  }
  return <InteropM2MVoucherInstructionsStep3 {...props} />
}

const ClientVoucherInstructionsStep3: React.FC<VoucherInstructionsStepProps> = ({
  purpose,
  back,
}) => {
  const { t } = useTranslation('voucher')

  const { data: descriptor, isLoading: isLoadingDescriptor } =
    EServiceQueries.useGetDescriptorCatalog(
      purpose?.eservice.id as string,
      purpose?.eservice.descriptor.id as string,
      { enabled: !!(purpose?.eservice.id && purpose?.eservice.descriptor.id), suspense: false }
    )

  const descriptorAudience = descriptor && descriptor.audience[0]

  return (
    <SectionContainer
      title={t('step3.consumer.title')}
      description={t('step3.consumer.description')}
    >
      <Stack spacing={4} sx={{ my: 4 }}>
        <InformationContainer
          label={t('step3.consumer.audField.label')}
          labelDescription={t('step3.consumer.audField.description')}
          content={
            <>
              {isLoadingDescriptor && !descriptorAudience && <Skeleton width={200} />}
              {descriptorAudience}
            </>
          }
          copyToClipboard={
            descriptorAudience
              ? {
                  value: descriptorAudience,
                  tooltipTitle: t('step3.consumer.audField.copySuccessFeedbackText'),
                }
              : undefined
          }
        />

        {purpose && (
          <InformationContainer
            label={t('step3.consumer.eserviceDetailsField.label')}
            content={
              <RouterLink
                to="SUBSCRIBE_CATALOG_VIEW"
                params={{
                  eserviceId: purpose.eservice.id,
                  descriptorId: purpose.eservice.descriptor.id,
                }}
                target="_blank"
              >
                {purpose.eservice.name}
              </RouterLink>
            }
          />
        )}
      </Stack>

      <StepActions back={{ label: t('backBtn'), type: 'button', onClick: back }} />
    </SectionContainer>
  )
}

const InteropM2MVoucherInstructionsStep3: React.FC<VoucherInstructionsStepProps> = ({ back }) => {
  const { t } = useTranslation('voucher')

  return (
    <SectionContainer title={t('step3.api.title')} description={t('step3.api.description')}>
      <InformationContainer
        sx={{ my: 4 }}
        label={t('step3.api.apiField.label')}
        content={
          <Link
            href={API_GATEWAY_INTEFACE_URL}
            target="_blank"
            rel="noreferrer"
            title={t('step3.api.apiField.link.title')}
          >
            {t('step3.api.apiField.link.label')}
          </Link>
        }
      />
      <StepActions back={{ label: t('backBtn'), type: 'button', onClick: back }} />
    </SectionContainer>
  )
}
