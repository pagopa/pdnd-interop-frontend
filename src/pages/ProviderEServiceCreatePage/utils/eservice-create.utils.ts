import type {
  AgreementApprovalPolicy,
  DescriptorAttributes,
  ProducerEServiceDescriptor,
} from '@/api/api.generatedTypes'
import type { RemappedEServiceAttributes } from '@/types/attribute.types'
import { getKeys } from '@/utils/array.utils'
import { remapEServiceAttributes } from '@/utils/attribute.utils'
import isEqual from 'lodash/isEqual'

type DescriptorDataStep2ToCompare = {
  voucherLifespan: number
  audience: string[]
  agreementApprovalPolicy: AgreementApprovalPolicy
  version: string
  description: string
  dailyCallsPerConsumer: number
  dailyCallsTotal: number
}

export function compareDescriptorsStep2DataIfEquals(
  newDescriptorData: DescriptorDataStep2ToCompare,
  descriptorToCompare: ProducerEServiceDescriptor
) {
  const descriptorDataToCompare: DescriptorDataStep2ToCompare = {
    voucherLifespan: descriptorToCompare.voucherLifespan,
    audience: descriptorToCompare.audience,
    agreementApprovalPolicy: descriptorToCompare.agreementApprovalPolicy,
    version: descriptorToCompare.version,
    description: descriptorToCompare.description ?? '',
    dailyCallsPerConsumer: descriptorToCompare.dailyCallsPerConsumer,
    dailyCallsTotal: descriptorToCompare.dailyCallsTotal,
  }

  return getKeys(newDescriptorData).every((key) =>
    isEqual(newDescriptorData[key], descriptorDataToCompare[key])
  )
}

export function compareAttributesStep3IfEquals(
  newAttributes: RemappedEServiceAttributes,
  attributesToCompare: DescriptorAttributes
) {
  const remappedAttributesToCompare = remapEServiceAttributes(attributesToCompare)

  const newAttributesToCompare = {
    certified: newAttributes.certified.filter((group) => group.attributes.length > 0),
    verified: newAttributes.verified.filter((group) => group.attributes.length > 0),
    declared: newAttributes.declared.filter((group) => group.attributes.length > 0),
  }

  return getKeys(newAttributes).every((key) =>
    isEqual(newAttributesToCompare[key], remappedAttributesToCompare[key])
  )
}
