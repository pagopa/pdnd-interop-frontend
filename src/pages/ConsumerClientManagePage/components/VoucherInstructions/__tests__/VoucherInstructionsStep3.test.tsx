import React from 'react'
import type { VoucherInstructionsStepProps } from '@/pages/ConsumerClientManagePage/types/voucher-instructions.types'
import { mockUseClientKind, renderWithApplicationContext } from '@/utils/testing.utils'
import { createMockPublicKey } from '__mocks__/data/key.mocks'
import { createMockPurpose } from '__mocks__/data/purpose.mocks'
import { vi } from 'vitest'
import { VoucherInstructionsStep3 } from '../VoucherInstructionsStep3'
import type { CatalogEServiceDescriptor } from '@/api/api.generatedTypes'
import { EServiceQueries } from '@/api/eservice'
import { createMockEServiceDescriptorCatalog } from '__mocks__/data/eservice.mocks'

const commonProps: VoucherInstructionsStepProps = {
  clientId: 'test-client-id',
  clientKeys: {
    keys: [
      createMockPublicKey({ keyId: '1', name: 'key1' }),
      createMockPublicKey({ keyId: '2', name: 'key2' }),
    ],
  },
  purpose: createMockPurpose(),
  purposeId: 'test-purpose-id',
  forward: vi.fn(),
  back: vi.fn(),
}

const mockUseGetDescriptorCatalog = (descriptor?: CatalogEServiceDescriptor, isLoading = false) => {
  vi.spyOn(EServiceQueries, 'useGetDescriptorCatalog').mockReturnValue({
    data: descriptor,
    isLoading,
  } as unknown as ReturnType<typeof EServiceQueries.useGetDescriptorCatalog>)
}

describe('VoucherInstructionsStep3', () => {
  it('should match snapshot (API)', () => {
    mockUseClientKind('API')
    const screen = renderWithApplicationContext(<VoucherInstructionsStep3 {...commonProps} />, {
      withRouterContext: true,
    })
    expect(screen.baseElement).toMatchSnapshot()
  })

  it('should match snapshot (CONSUMER)', () => {
    mockUseClientKind('CONSUMER')
    mockUseGetDescriptorCatalog(createMockEServiceDescriptorCatalog())
    const screen = renderWithApplicationContext(<VoucherInstructionsStep3 {...commonProps} />, {
      withRouterContext: true,
    })
    expect(screen.baseElement).toMatchSnapshot()
  })

  it('should match snapshot (CONSUMER) while loading descriptor', () => {
    mockUseClientKind('CONSUMER')
    mockUseGetDescriptorCatalog(undefined, true)
    const screen = renderWithApplicationContext(<VoucherInstructionsStep3 {...commonProps} />, {
      withRouterContext: true,
    })
    expect(screen.baseElement).toMatchSnapshot()
  })
})
