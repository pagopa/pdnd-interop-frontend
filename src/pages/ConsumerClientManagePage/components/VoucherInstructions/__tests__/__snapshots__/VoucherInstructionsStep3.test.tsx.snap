// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VoucherInstructionsStep3 > should match snapshot (API) 1`] = `
<body>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1w5x330-MuiPaper-root"
    >
      <div
        class="css-nen11g-MuiStack-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
        >
          step3.api.title
        </h2>
        <span
          class="MuiTypography-root MuiTypography-caption css-ol7wxs-MuiTypography-root"
        >
          step3.api.description
        </span>
      </div>
      <div
        class="MuiBox-root css-1yuhvjn"
      >
        <div
          class="css-1evo70x-MuiStack-root"
        >
          <div
            class="MuiBox-root css-iuqs4u"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
              step3.api.apiField.label
            </p>
          </div>
          <div
            class="MuiBox-root css-1rr4qq7"
          >
            <div
              class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
                href="https://test.it//swagger/docs/interface-specification.yml"
                rel="noreferrer"
                target="_blank"
                title="step3.api.apiField.link.title"
              >
                step3.api.apiField.link.label
              </a>
            </div>
          </div>
        </div>
        <div
          class="css-9v4aab-MuiStack-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            backBtn
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`VoucherInstructionsStep3 > should match snapshot (CONSUMER) 1`] = `
<body>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1w5x330-MuiPaper-root"
    >
      <div
        class="css-nen11g-MuiStack-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
        >
          step3.consumer.title
        </h2>
        <span
          class="MuiTypography-root MuiTypography-caption css-ol7wxs-MuiTypography-root"
        >
          step3.consumer.description
        </span>
      </div>
      <div
        class="MuiBox-root css-1yuhvjn"
      >
        <div
          class="css-vemggx-MuiStack-root"
        >
          <div
            class="css-19qjufr-MuiStack-root"
          >
            <div
              class="MuiBox-root css-iuqs4u"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
              >
                step3.consumer.audField.label
              </p>
              <hr
                class="MuiDivider-root MuiDivider-fullWidth css-159wgzx-MuiDivider-root"
              />
              <span
                class="MuiTypography-root MuiTypography-caption css-g171c3-MuiTypography-root"
              >
                step3.consumer.audField.description
              </span>
            </div>
            <div
              class="MuiBox-root css-1rr4qq7"
            >
              <div
                class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
              >
                Lorem
                <button
                  aria-label="Copia"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-vqczhe-MuiButtonBase-root-MuiIconButton-root"
                  data-mui-internal-clone-element="true"
                  role="button"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                    data-testid="ContentCopyIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
          <div
            class="css-19qjufr-MuiStack-root"
          >
            <div
              class="MuiBox-root css-iuqs4u"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
              >
                step3.consumer.eserviceDetailsField.label
              </p>
            </div>
            <div
              class="MuiBox-root css-1rr4qq7"
            >
              <div
                class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
              >
                <a
                  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
                  href="/it/fruizione/catalogo-e-service/dea4bbf4-df64-4b8a-9ca9-125dd4cd1f5e/cd80bfe6-54be-493a-aaa1-6bb2b54545f8"
                  target="_blank"
                >
                  Test Attributi 2 - Ste
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="css-9v4aab-MuiStack-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            backBtn
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`VoucherInstructionsStep3 > should match snapshot (CONSUMER) while loading descriptor 1`] = `
<body>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1w5x330-MuiPaper-root"
    >
      <div
        class="css-nen11g-MuiStack-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
        >
          step3.consumer.title
        </h2>
        <span
          class="MuiTypography-root MuiTypography-caption css-ol7wxs-MuiTypography-root"
        >
          step3.consumer.description
        </span>
      </div>
      <div
        class="MuiBox-root css-1yuhvjn"
      >
        <div
          class="css-vemggx-MuiStack-root"
        >
          <div
            class="css-19qjufr-MuiStack-root"
          >
            <div
              class="MuiBox-root css-iuqs4u"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
              >
                step3.consumer.audField.label
              </p>
              <hr
                class="MuiDivider-root MuiDivider-fullWidth css-159wgzx-MuiDivider-root"
              />
              <span
                class="MuiTypography-root MuiTypography-caption css-g171c3-MuiTypography-root"
              >
                step3.consumer.audField.description
              </span>
            </div>
            <div
              class="MuiBox-root css-1rr4qq7"
            >
              <div
                class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                  style="width: 200px;"
                />
              </div>
            </div>
          </div>
          <div
            class="css-19qjufr-MuiStack-root"
          >
            <div
              class="MuiBox-root css-iuqs4u"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
              >
                step3.consumer.eserviceDetailsField.label
              </p>
            </div>
            <div
              class="MuiBox-root css-1rr4qq7"
            >
              <div
                class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
              >
                <a
                  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
                  href="/it/fruizione/catalogo-e-service/dea4bbf4-df64-4b8a-9ca9-125dd4cd1f5e/cd80bfe6-54be-493a-aaa1-6bb2b54545f8"
                  target="_blank"
                >
                  Test Attributi 2 - Ste
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="css-9v4aab-MuiStack-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            backBtn
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
